pkg/virtctl/spice/spice.go

```diff
@@ -0,0 +1,95 @@
+package spice
+
+import (
+	"errors"
+	"fmt"
+	flag "github.com/spf13/pflag"
+	"io/ioutil"
+	kubev1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/rest"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"log"
+	"os"
+	"os/exec"
+)
+
+const FLAG = "spice"
+const SPICE_FILENAME = "console.vv"
+const VM_SPICE_URL = "http://%s/apis/kubevirt.io/v1alpha1/namespaces/default/vms/%s/spice"
+
+type Spice struct {
+}
+
+func DownloadSpice(vm string, restClient *rest.RESTClient) (string, error) {
+	body, err := restClient.Get().
+		Resource("vms").SetHeader("Accept", "text/plain").
+		SubResource("spice").
+		Namespace(kubev1.NamespaceDefault).
+		Name(vm).Do().Raw()
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Can't read body: %s\n", err.Error()))
+	}
+	return fmt.Sprintf("%s", body), nil
+}
+
+func (o *Spice) FlagSet() *flag.FlagSet {
+
+	cf := flag.NewFlagSet(FLAG, flag.ExitOnError)
+	cf.BoolP("details", "d", false, "If present, print SPICE console to stdout, otherwise run remote-viewer")
+	return cf
+}
+
+func (o *Spice) Run(flags *flag.FlagSet) int {
+	server, _ := flags.GetString("server")
+	kubeconfig, _ := flags.GetString("kubeconfig")
+	details, _ := flags.GetBool("details")
+
+	if len(flags.Args()) != 2 {
+		log.Println("VM name is missing")
+		return 1
+	}
+	vm := flags.Arg(1)
+
+	restClient, err := kubecli.GetRESTClientFromFlags(server, kubeconfig)
+
+	if err != nil {
+		log.Println(err)
+		return 1
+	}
+	body, err := DownloadSpice(vm, restClient)
+	if err != nil {
+		log.Fatalf(err.Error())
+		return 1
+	}
+	if details {
+		fmt.Printf("%s", body)
+	} else {
+		f, err := ioutil.TempFile("", "spice")
+
+		if err != nil {
+			log.Fatalf("Can't open file: %s", err.Error())
+			return 1
+		}
+		defer f.Close()
+
+		_, err = f.WriteString(body)
+		if err != nil {
+			log.Fatalf("Can't write to file: %s", err.Error())
+			return 1
+		}
+
+		f.Sync()
+
+		cmnd := exec.Command("remote-viewer", f.Name())
+		cmnd.Run()
+		os.Remove(f.Name())
+	}
+	return 0
+}
+
+func (o *Spice) Usage() string {
+	usage := "Usage: ./virtctl spice vm_name [--details] [common_options]\n\n"
```

Comment:
Could you give real examples of the usage here? For instance kubectl exec --help gives:

```
Execute a command in a container.

Examples:
  # Get output from running 'date' from pod 123456-7890, using the first container by default
  kubectl exec 123456-7890 date
  
  # Get output from running 'date' in ruby-container from pod 123456-7890
  kubectl exec 123456-7890 -c ruby-container date
  
  # Switch to raw terminal mode, sends stdin to 'bash' in ruby-container from pod 123456-7890
  # and sends stdout/stderr from 'bash' back to the client
  kubectl exec 123456-7890 -c ruby-container -i -t -- bash -il

Options:
  -c, --container='': Container name. If omitted, the first container in the pod will be chosen
  -p, --pod='': Pod name
  -i, --stdin=false: Pass stdin to the container
  -t, --tty=false: Stdin is a TTY

Usage:
  kubectl exec POD [-c CONTAINER] -- COMMAND [args...] [options]
```

Maybe

```
Fetch SPICE connections details of a VM and connect via remote-viewer

Examples:
#Show connection details of the VM myvm
virtctl spice myvm --details

# Open a SPICE connection to the VM myvm via remote-viewer
virtctl spice myvm

Options:
```
The rest will be added by o.FlagSet.FlagUsages()